pkg/messaging/mqtt/pubsub.go

```diff
@@ -191,6 +193,23 @@ func newClient(address, id string, timeout time.Duration) (mqtt.Client, error) {
 	return client, nil
 }
 
+func (ps *pubsub) mqttHandler(h messaging.MessageHandler) mqtt.MessageHandler {
+	return func(c mqtt.Client, m mqtt.Message) {
+		var msg messaging.Message
+		if err := proto.Unmarshal(m.Payload(), &msg); err != nil {
```

Comment:
If unmarshalling fails, the payload is likely either empty or messed up. Either way, no need to continue with `Handle`.